queries/Query3.sql

```diff
@@ -0,0 +1,30 @@
+-- Borrowing Frequency using Window Functions: Rank borrowers based on borrowing frequency.
+
+USE [LibraryManagement]
+GO
+
+WITH BorrowingFrequency AS (
+    SELECT 
+        br.BorrowerID,
+        br.FirstName,
+        br.LastName,
+        COUNT(l.LoanID) AS TotalBorrowed,
+        ROW_NUMBER() OVER (ORDER BY COUNT(l.LoanID) DESC) AS BorrowerRank
```

Comment:
Here you can use `DENSE_RANK() OVER (ORDER BY COUNT(l.LoanID) DESC) AS BorrowerRank`
it's more appropriate when ranking by borrowing frequency (no gaps)